pandas/core/index.py

```diff
@@ -1411,7 +1411,11 @@ def shift(self, periods=1, freq=None):
             # OK because immutable
             return self
 
-        offset = periods * freq
+        try:
+            offset = periods * freq
+        except TypeError:
```

Comment:
no, pls. raise a ``NotImplementedError`` for all ops in ``core/index.py`` for ``.shift``. Then define this in ``tseries/base/DatetimeIndexOpsMixIn``